app/models/mixins/authentication_mixin.rb

```diff
@@ -488,8 +485,9 @@ def authentication_component(type, method)
   end
 
   def available_authentications
-    authentication_userid_passwords + authentication_key_pairs + authentication_tokens
+    authentications.where.not(:authtype => nil)
```

Comment:
aka - what you did here - nice

Reply:
Yeah we also had this as it's own association with a scope but a number of places depended on accessing the in-memory association cache before it was persisted